static/scripts/app/views/dir.js

```diff
@@ -50,6 +51,45 @@ define([
                         tNext: gettext("Next (Right arrow key)"), // Alt text on right arrow
                         tCounter: gettext("%curr% of %total%") // Markup for "1 of 7" counter
                     },
+                    callbacks: {
+                        elementParse: function(item) {
+                            var repo_id = _this.dir.repo_id,
+                                cur_path = _this.dir.path,
+                                cur_img = _this.dir.find(function(dirent) {
+                                    return dirent.get('obj_name') == item.el[0].innerText;
```

Comment:
I think I'll take a look into it again and add the obj_id as data-attribute to the <a> element. this way we don't run into trouble with character sets and other filename issues. Is likely the cleaner way.